packages/connectivity/src/scp-cf/environment-accessor/service-credentials.ts

```diff
@@ -94,7 +93,8 @@ function getCredentialsWithJwt<ServiceCredentialsT extends ServiceCredentials>(
     c => matchesClientId(c, token) || matchesAudience(c, token)
   );
 
-  logResult(service, eligibleCredentials, true);
+  // eslint-disable-next-line no-unused-expressions
+  eligibleCredentials[0] ? logResult(service, eligibleCredentials, true) : logger.warn(`Found no service binding for service '${service}' matching either the token's client id or audience.`);
```

Comment:
[req|pp] Nooooooo! 😱 🤪

Please just add this to `logResult()` instead 😊

Reply:
This was intentional, the logResult() method is used for both `getCredentialsWithoutJwt` and `getCredentialsWithJwt`, this specific logic only applies to `...WithJwt`.

Reply:
Now that I look at it again, we already pass a "usedToken" boolean, I'm an idiot 🤦